src/Discord/Internal/Gateway/Cache.hs

```diff
@@ -20,25 +20,19 @@ data Cache = Cache
      , cacheDMChannels :: !(M.Map ChannelId Channel)
      , cacheGuilds :: !(M.Map GuildId (Maybe (Guild, Maybe GuildCreateData)))
      , cacheChannels :: !(M.Map ChannelId Channel)
-     , cacheApplication :: !PartialApplication
+     , cacheApplication :: !FullApplication
      } deriving (Show)
 
 data CacheHandle = CacheHandle
   { cacheHandleEvents :: Chan (Either GatewayException EventInternalParse)
   , cacheHandleCache  :: MVar (Either (Cache, GatewayException) Cache)
```

Comment:
Like I did in my branch I don't think there's anything useful about storing the gatewayexception here alongside the cache. In fact, in the `loop` if we ever get a gateway exception we refuse to update the cache, which is a mistake imo

Reply:
This plus your `when isEnabled loop` suggestion greatly simplified `cacheLoop`